ui/app/modifiers/code-mirror.ts

```diff
@@ -0,0 +1,59 @@
+import Modifier from 'ember-modifier';
+import codemirror from 'codemirror';
+
+import waypointHclMode from './utils/waypointHclMode';
+import 'codemirror/addon/edit/matchbrackets';
+import 'codemirror/addon/edit/closebrackets';
+import 'codemirror/addon/selection/active-line';
+import 'codemirror/addon/mode/simple';
+
+const _PRESET_DEFAULTS: codemirror.EditorConfiguration = {
+  theme: 'monokai',
+  lineNumbers: true,
+  cursorBlinkRate: 500,
+  matchBrackets: true,
+  autoCloseBrackets: true,
+  styleActiveLine: true,
+};
+interface Args {
+  positional: never;
+  named: {
+    value: string;
+    onInput: (value: string) => void;
+    options: Record<string, unknown>;
```

Comment:
I reckon it’s worth marking this as optional to keep us out of trouble in future.

```suggestion
    options?: Record<string, unknown>;
```